runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/GrpcCommitWorkStream.java

```diff
@@ -314,26 +418,43 @@ private Batcher() {
     @Override
     public boolean commitWorkItem(
         String computation, WorkItemCommitRequest commitRequest, Consumer<CommitStatus> onDone) {
+      if (isShutdown()) {
+        onDone.accept(CommitStatus.ABORTED);
```

Comment:
if we return false I don't think the caller expects onDone to be called.

Reply:
done